public/app/plugins/datasource/jaeger/components/QueryEditor.tsx

```diff
@@ -1,38 +1,24 @@
 import { css } from '@emotion/css';
 import { QueryEditorProps } from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
-import { InlineField, InlineFieldRow, Input, RadioButtonGroup } from '@grafana/ui';
+import { FileDropzone, InlineField, InlineFieldRow, Input, RadioButtonGroup, useTheme2 } from '@grafana/ui';
 import React from 'react';
 import { JaegerDatasource } from '../datasource';
 import { JaegerQuery, JaegerQueryType } from '../types';
 import { SearchForm } from './SearchForm';
 
 type Props = QueryEditorProps<JaegerDatasource, JaegerQuery>;
 
-export function QueryEditor({ datasource, query, onChange }: Props) {
-  return (
-    <div className={css({ width: '50%' })}>
-      <InlineFieldRow>
-        <InlineField label="Query type">
-          <RadioButtonGroup<JaegerQueryType>
-            options={[
-              { value: 'search', label: 'Search' },
-              { value: undefined, label: 'TraceID' },
-            ]}
-            value={query.queryType}
-            onChange={(v) =>
-              onChange({
-                ...query,
-                queryType: v,
-              })
-            }
-            size="md"
-          />
-        </InlineField>
-      </InlineFieldRow>
-      {query.queryType === 'search' ? (
-        <SearchForm datasource={datasource} query={query} onChange={onChange} />
-      ) : (
+export function QueryEditor({ datasource, query, onChange, onRunQuery }: Props) {
+  const theme = useTheme2();
+
+  const renderEditorBody = () => {
```

Comment:
Would it make sense to render all 3 options (search, TraceID, upload) from the same function using a switch statement or chained ternaries in the function return? Might be easier to track the different UI's being shown.

Reply:
Oh yeah that was my original way but then I extracted the upload view because it needs 100% width but your comment made me realise that I can still render the upload inside that function and change the width when upload is the query type. Thanks